grapheneapi/graphenews.py

```diff
@@ -179,16 +181,21 @@ def run_forever(self) :
                     coro = loop.create_connection(self.factory, self.host,
                                                   self.port, ssl=context)
                 else :
+
                     coro = loop.create_connection(self.factory, self.host,
                                                   self.port, ssl=self.ssl)
-
                 loop.run_until_complete(coro)
                 loop.run_forever()
             except KeyboardInterrupt:
                 break
 
-            print("Trying to re-connect in 10 seconds!")
-            time.sleep(10)
-
+            try:
+                print("Multiple witnesses configured, trying another witness.")
+                GrapheneWebsocketRPC.__init__(self, self.url_list, self.username, self.password)
+                self.ssl, self.host, self.port, self.resource, self.path, self.params = parseWsUrl(self.url)
+                self.connect()
```

Comment:
`self.connect()` is not enough you would need to call `def run_forever()` again which might end up in an infinite loop.
`loop.run_forever()` is what starts the event loop and awaits data from websocket.

Maybe move the reconnect into the while loop and have a counter that is reset on successful connection.